packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -145,8 +152,13 @@ export class ModuleGraph {
     }
     // Don't invalidate mod.info and mod.meta, as they are part of the processing pipeline
     // Invalidating the transform result is enough to ensure this module is re-processed next time it is requested
-    mod.transformResult = null
-    mod.ssrTransformResult = null
+    if (softInvalidate && mod.softInvalidated !== false) {
+      mod.softInvalidated = true
+    } else {
+      mod.softInvalidated = false
+      mod.transformResult = null
+      mod.ssrTransformResult = null
+    }
     mod.ssrModule = null
     mod.ssrError = null
```

Comment:
Don't know how it's used but maybe it should be reset only in combination with `ssrTransformResult`?

Reply:
Do you mean `ssrModule` and `ssrError`? From what I can track, they are used around here:

https://github.com/vitejs/vite/blob/169a3fff969a2991a3a78eb95890aa23f149cb84/packages/vite/src/node/ssr/ssrModuleLoader.ts#L85-L107

And ideally we should set `ssrModule` and `ssrError` to null as they're kinda the equivalent of timestamps for SSR. In the code above, `transformRequest` would still get the fast path we want, but `ssrModule` would be instantiated with a new instance, which I _think_ is what we want.